File: app/src/main/java/com/example/prueba001/fragments/list/adapter/ListFragmentAdapter.java
package com.example.prueba001.fragments.list.adapter;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.bumptech.glide.Glide;
import com.bumptech.glide.load.resource.bitmap.CircleCrop;
import com.example.prueba001.R;
import com.example.prueba001.databinding.FragmentHeroListItemBinding;
import com.example.prueba001.model.HeroModel;

import java.util.ArrayList;
import java.util.List;

public class ListFragmentAdapter extends RecyclerView.Adapter<RecyclerView.ViewHolder> {

    private List<HeroModel> heroList;
    private Context context;
    private OnHeroClickCallback callback;

    public ListFragmentAdapter(OnHeroClickCallback callback, Context context) {
        heroList = new ArrayList<>();
        this.context = context;
        this.callback = callback;
    }

    public void setHeroList(List<HeroModel> heroList) {
        this.heroList = heroList;
    }

    @NonNull
    @Override
    public RecyclerView.ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        FragmentHeroListItemBinding fragmentHeroListItemBinding = FragmentHeroListItemBinding.inflate(LayoutInflater.from(parent.getContext()), parent, false);
        return new HeroViewHolder(fragmentHeroListItemBinding);
    }

    @Override
    public void onBindViewHolder(@NonNull RecyclerView.ViewHolder holder, int position) {
        HeroViewHolder heroViewHolder = (HeroViewHolder) holder;
        heroViewHolder.bind(heroList.get(position));
    }

    @Override
    public int getItemCount() {
        return heroList.size();
    }

    private class HeroViewHolder extends RecyclerView.ViewHolder {

        private FragmentHeroListItemBinding binding;

        public HeroViewHolder(@NonNull FragmentHeroListItemBinding itemView) {
            super(itemView.getRoot());
            this.binding = itemView;
        }

        public void bind(HeroModel hero) {
            if (hero.getImages() != null) {
                Glide.with(context)
                        .load(hero.getImages().getLg())
                        .transform(new CircleCrop())
                        .into(binding.ivHero);
            }

            binding.tvName.setText(hero.getName());

            if (hero.getPowerstats() != null) {

                binding.tvIntelligence.setText(context.getString(R.string.hero_alignment) + hero.getPowerstats().getIntelligence());

                binding.tvStrenght.setText(context.getString(R.string.hero_strength) + hero.getPowerstats().getStrength());

                binding.tvSpeed.setText(context.getString(R.string.hero_speed) + hero.getPowerstats().getSpeed());

                binding.tvDurability.setText(context.getString(R.string.hero_durability) + hero.getPowerstats().getDurability());

                binding.tvPower.setText(context.getString(R.string.hero_power) + hero.getPowerstats().getPower());

                binding.tvCombat.setText(context.getString(R.string.hero_combat) + hero.getPowerstats().getCombat());
            }

            if (hero.isFavorite()) {
                binding.ivFav.setVisibility(View.VISIBLE);
            } else {
                binding.ivFav.setVisibility(View.GONE);
            }

            binding.container.setOnClickListener(new View.OnClickListener() {
                @Override
                public void onClick(View view) {
                    callback.onHeroClick(hero);
                }
            });
        }

    }

}

File: app/src/main/java/com/example/prueba001/fragments/list/adapter/ListFragmentAdapter.kt
package com.example.prueba001.fragments.list.adapter

import android.content.Context
import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.appcompat.content.res.AppCompatResources
import androidx.recyclerview.widget.RecyclerView
import com.bumptech.glide.Glide
import com.bumptech.glide.load.resource.bitmap.CircleCrop
import com.example.prueba001.R
import com.example.prueba001.databinding.FragmentHeroListItemBinding
import com.example.prueba001.model.HeroModel

class ListFragmentAdapter(callback: OnHeroClickCallback, context: Context) :
    RecyclerView.Adapter<RecyclerView.ViewHolder>() {

    var heroList: List<HeroModel> = emptyList()
    private val context: Context
    private val callback: OnHeroClickCallback

    init {
        this.context = context
        this.callback = callback
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): RecyclerView.ViewHolder {
        val fragmentHeroListItemBinding = FragmentHeroListItemBinding.inflate(LayoutInflater.from(parent.context), parent, false)
        return HeroViewHolder(fragmentHeroListItemBinding)
    }

    override fun onBindViewHolder(holder: RecyclerView.ViewHolder, position: Int) {
        val heroViewHolder = holder as HeroViewHolder
        heroViewHolder.bind(heroList[position])
    }

    override fun getItemCount(): Int {
        return heroList.size
    }

    private inner class HeroViewHolder(private val binding: FragmentHeroListItemBinding) :
        RecyclerView.ViewHolder(binding.root) {

        fun bind(hero: HeroModel) {
            hero.images?.let {
                Glide.with(context)
                    .load(it.lg)
                    .transform(CircleCrop())
                    .into(binding.ivHero)
            }
            binding.tvName.text = hero.name
            hero.powerstats?.let { stats ->
                with(binding) {
                    tvIntelligence.text = context.getString(R.string.hero_alignment) + stats.intelligence
                    tvStrenght.text = context.getString(R.string.hero_strength) + stats.strength
                    tvSpeed.text = context.getString(R.string.hero_speed) + stats.speed
                    tvDurability.text = context.getString(R.string.hero_durability) + stats.durability
                    tvPower.text = context.getString(R.string.hero_power) + stats.power
                    tvCombat.text = context.getString(R.string.hero_combat) + stats.combat
                }
            }
            if (hero.isFavorite) {
                binding.ivFav.background = AppCompatResources.getDrawable(context, R.drawable.ic_fav)
            } else {
                binding.ivFav.background = AppCompatResources.getDrawable(context, R.drawable.ic_unfav)
            }
            binding.container.setOnClickListener { callback.onHeroClick(hero) }
            binding.ivFav.setOnClickListener { callback.onFavChanged(hero) }
        }
    }
}